### DISCUSSION TITLE
Installation on nginx 1.22.1

### DISCUSSION THREAD

**Discussion by schmid786** [firefly-iii/firefly-iii#6730] [Firefly III]
I'm trying to install firefly-iii on my homelab with nginx version 1.22.1 and php8.1, but I couldn't get it running.
I used the server configuration from the FAQ site with no success so far. 
Without modifications I get an "403 Forbidden" message.
If I add an index directive to the budget location (`index index.php;`) get redirected to "budget/login" which shows only the message "Access denied."
The nginx error log contains:
<code>2022/12/20 21:47:22 [error] 3912#3912: *78 FastCGI sent in stderr: "Access to the script '/usr/share/nginx/firefly-iii/public/index.php/login' has been denied (see security.limit_extensions)" while reading response header from upstream, client: ..., server: ..., request: "GET /budget/login HTTP/1.1", upstream: "fastcgi://unix:/run/php/php8.1-fpm.sock:", host: "..."
</code>
I think that the script name is wrong as the "/login" part should only be in the PATH_INFO parameter but I have no clue how to accomplish that.

**Comment by JC5** [comment]
Could you share your current configuration?

**Comment by schmid786** [comment]
My server config is as follows. I have also a static website (`/`) and phpMyAdmin (`/phpMyAdmin`) running on this server, both working fine.

    server {
        listen       80 default_server;
        server_name  <local_server_ip>;

        access_log  /var/log/nginx/local.access.log  main;

        location / {
            root   /usr/share/nginx/html;
            index  index.html index.htm;
        }

        location /phpMyAdmin {
            root   /usr/share/nginx;
            index  index.php index.html;
            allow  <local_ip_range>;
            deny   all;
        }

        location ~ /phpMyAdmin/.+\.php$ {
            root /usr/share/nginx;
            include fastcgi_params;
            fastcgi_index index.php;
            fastcgi_split_path_info ^(.+\.php)(/.+)$;
            fastcgi_intercept_errors on;
            fastcgi_pass unix:/run/php/php-fpm.sock;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        }

        location ^~ /firefly-iii/ {
            deny all;
        }

        location ^~ /budget {
            alias /usr/share/nginx/firefly-iii/public;
            index index.php;
            try_files $uri $uri/ @budget;

            location ~* \.php(?:$|/) {
                include fastcgi_params;
                fastcgi_param SCRIPT_FILENAME $request_filename;
                fastcgi_param modHeadersAvailable true;
                fastcgi_pass unix:/run/php/php8.1-fpm.sock;
            }
        }

        location @budget {
            rewrite ^/budget/(.*)$ /budget/index.php/$1 last;
        }

        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
            root   /usr/share/nginx/html;
        }
    }

I have a MariaDB instance running on a seperate machine on my local network and edited the `.env` file accordingly. This seems to work as the intitialization of the database works and the tables are created.

I had in my php.ini the value `cgi.fix_pathinfo=0;` set, which is not neseccary, I think. But now with the default value (`1`) I get a 404 Error from FireflyIII.

    FireflyIII 404 Not Found :(
    404 — Firefly III cannot find this page.

**Comment by schmid786** [comment]
I got it running by adding `fastcgi_split_path_info ^(.+\.php)(/.+)$;` into the php-location.

**Reply by JC5** [reply]
Nice!
